google-beta/resource_compute_disk.go

```diff
@@ -773,20 +812,29 @@ func resourceComputeDiskDelete(d *schema.ResourceData, meta interface{}) error {
 			}
 		}
 		for _, call := range detachCalls {
-			op, err := config.clientCompute.Instances.DetachDisk(call.project, call.zone, call.instance, call.deviceName).Do()
-			if err != nil {
-				return fmt.Errorf("Error detaching disk %s from instance %s/%s/%s: %s", call.deviceName, call.project,
-					call.zone, call.instance, err.Error())
-			}
-			err = computeOperationWait(config.clientCompute, op, call.project,
-				fmt.Sprintf("Detaching disk from %s/%s/%s", call.project, call.zone, call.instance))
-			if err != nil {
-				if opErr, ok := err.(ComputeOperationError); ok && len(opErr.Errors) == 1 && opErr.Errors[0].Code == "RESOURCE_NOT_FOUND" {
-					log.Printf("[WARN] instance %q was deleted while awaiting detach", call.instance)
-					continue
+			err = resource.Retry(time.Duration(5)*time.Minute, func() *resource.RetryError {
```

Comment:
I'm not familiar with timeouts, but should this length be pulled from those instead of hardcoded?

Reply:
Short answer, yes but this is a place holder until that work is finished. See https://github.com/GoogleCloudPlatform/magic-modules/pull/985/files#diff-b12861b106185324fa6bfdb16cbf1834R73 and the associated PR for more